internal/daemonset/daemonset.go

```diff
@@ -52,12 +53,13 @@ func NewCreator(client client.Client, kernelLabel string, scheme *runtime.Scheme
 	}
 }
 
-func (dc *daemonSetGenerator) GarbageCollect(ctx context.Context, existingDS map[string]*appsv1.DaemonSet, validKernels sets.Set[string]) ([]string, error) {
+func (dc *daemonSetGenerator) GarbageCollect(ctx context.Context, existingDS []appsv1.DaemonSet, validKernels sets.Set[string]) ([]string, error) {
 	deleted := make([]string, 0)
 
-	for kernelVersion, ds := range existingDS {
-		if !dc.isDevicePluginDaemonSet(ds) && !validKernels.Has(kernelVersion) {
-			if err := dc.client.Delete(ctx, ds); err != nil {
+	for _, ds := range existingDS {
+		dsKernelVersion := ds.Labels[dc.kernelLabel]
+		if !IsDevicePluginDS(&ds) && !validKernels.Has(dsKernelVersion) {
```

Comment:
```suggestion
		
		if dsKernelVersion := ds.Labels[dc.kernelLabel]; !IsDevicePluginDS(&ds) && !validKernels.Has(dsKernelVersion) {
```

This variable is only scoped to the `if` block

Reply:
fixed